code/modules/atmospherics/environmental/LINDA_fire.dm

```diff
@@ -10,37 +10,31 @@
 
 
 /turf/open/hotspot_expose(exposed_temperature, exposed_volume, soh)
-	var/datum/gas_mixture/air_contents = return_air()
-	if(!air_contents)
-		return 0
+	var/list/air_gases = air?.gases
+	if(!air_gases)
+		return
 
-	var/oxy = air_contents.gases[/datum/gas/oxygen] ? air_contents.gases[/datum/gas/oxygen][MOLES] : 0
-	var/tox = air_contents.gases[/datum/gas/plasma] ? air_contents.gases[/datum/gas/plasma][MOLES] : 0
-	var/trit = air_contents.gases[/datum/gas/tritium] ? air_contents.gases[/datum/gas/tritium][MOLES] : 0
+	var/oxy = air_gases[/datum/gas/oxygen] ? air_gases[/datum/gas/oxygen][MOLES] : 0
```

Comment:
`air_gases[/datum/gas/oxygen]` is an associated lookup. Its worth it to cache the value rather than look it up twice for these checks if we are aiming for free real estate here.

Reply:
associated lookup vs. defining a new proc var that has to be cleaned up when the proc ends, i have no idea how that shakes out

Reply:
You technically would only need 1 proc var to get rid of 3 associated lookups.

The return value isn't used so you could also use `.` and avoid defining the var.